Build.PL

```diff
@@ -37,28 +36,48 @@ my $class = npg_tracking::util::build->subclass(code => <<'EOF');
   sub _samtools_option {
     my $self = shift;
 
-    my @found = which('samtools');
-    my $samtools = @found ? abs_path( $found[0] ) : q[];
-    @found = which('htsfile');
-    my $hts = @found ? abs_path( $found[0] ) : q[];
-    my $pkgs = [];
-    if ($samtools && $hts) {
-      foreach my $p (($samtools, $hts)) {
-        $p = dirname $p;
-        if ( $p =~ m{/bin\Z}smx ) {
-          $p = dirname $p;
-        }
-        if ($self->verbose) {
-          warn "Found $p\n";
-        }
-        push @{$pkgs}, $p;
-        $samtools = 'SAMTOOLS_LOC=' . $samtools;
+    my ($samtools, $htslib);
+
+    # Requires the samtools source code to build
+    if ($ENV{SAMTOOLS_SOURCE_PATH}) {
+      $samtools = $ENV{SAMTOOLS_SOURCE_PATH};
+    } else {
+      $samtools = _relative_to_bin('samtools')
+    }
+
+    if ($ENV{HTSLIB_INSTALL_PATH}) {
+      $htslib = $ENV{HTSLIB_INSTALL_PATH};
+    } else {
+      $htslib = _relative_to_bin('htsfile')
+    }
+
+    my @pkgs;
+    if ($samtools && $htslib) {
+      push @pkgs, "SAMTOOLS_LOC=$samtools";
+      push @pkgs, "HTSLIB_LOC=$htslib";
+
+      if ($self->verbose) {
+        warn "Found samtools source $samtools\n";
+        warn "Found htslib installed $htslib\n";
       }
-      $pkgs->[0] = 'SAMTOOLS_LOC=' . $pkgs->[0];
-      $pkgs->[1] = 'HTSLIB_LOC='   . $pkgs->[1];
     }
 
-    return $pkgs;
+    return \@pkgs;
+  }
+
+  sub _relative_to_bin {
+    my ($executable) = @_;
+
+    my $path;
+    my @found = which($executable);
+    if (@found) {
+      $path = abs_path(dirname $found[0]);
```

Comment:
Does not seem to work:
Found samtools source /software/npg/20170425
Found htslib installed /software/npg/20170425
make: Entering directory `/nfs/users/nfs_m/mg8/working/npg_qc/src/fastq_summ'
cc -c -g -Wall -I. -I/software/npg/20170425/include -I/software/npg/20170425/include fastq_summ.c -o build/fastq_summ.o
fastq_summ.c:71:17: fatal error: sam.h: No such file or directory
compilation terminated.
The following works:

$path = dirname abs_path($found[0]);